contracts/MetaGuard.sol

```diff
@@ -0,0 +1,187 @@
+// SPDX-License-Identifier: ISC
+pragma solidity ^0.8.9;
+pragma abicoder v1;
+
+import "@gnosis.pm/zodiac/contracts/interfaces/IAvatar.sol";
+import "@gnosis.pm/zodiac/contracts/guard/BaseGuard.sol";
+import "@gnosis.pm/zodiac/contracts/factory/FactoryFriendly.sol";
+
+contract MetaGuard is FactoryFriendly, BaseGuard {
+    address internal constant SENTINEL_GUARDS = address(0x1);
+
+    address public avatar;
+    mapping(address => address) internal guards;
```

Comment:
Why not AddressSet?

Reply:
Actually just following how zodiac manages modules, and I think with the linked list we have less code than AddressSet